### PR TITLE
feat: remove deprecated precision amounts

### PR DESCRIPTION
**Pull request by gomesalexandre**
## Description

Part 2 of base units refactor - DeFi still expects precision amounts within the abstraction, but those are now derived in-place at component-level.

This PR:

- removes the precision amounts that were marked as deprecated in https://github.com/shapeshift/web/pull/3409
- adds `precision` vernacular to the passed down props
- does the base unit to precision conversion in-place in components

This puts us in a sane place for a follow-up PR to 
- remove the in-place precision conversion
- have DeFi take base units
and use precision amounts as *display only*, while never doing back and forth conversion 

## Notice

- [x] Have you followed the guidelines in our [Contributing]('https://github.com/shapeshift/web/blob/develop/CONTRIBUTING.md) guide?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/shapeshift/web/pulls) for the same update/change?

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

N/A

## Risk

Low/medium - precision used could be the wrong one in some DeFi domain places

## Testing

- DeFi modals amount show no regression, test all opportunities, all steps, test max withdraws are still working. Fiat amounts should still look the same, and inputting crypto/fiat should show no regression on the mirrored field
- DeFi cards/rows amounts should show no regression, over all opportunities
- DeFi earn balance/net worth should show no regression

### Engineering

- Precision amounts have been consistently removed

### Operations

☝🏽 

## Screenshots (if applicable)

### CODE REVIEW

**Comment by gomesalexandre** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #3423** <a href="https://app.graphite.dev/github/pr/shapeshift/web/3423" target="_blank">Graphite</a> 
    * **PR #3424** <a href="https://app.graphite.dev/github/pr/shapeshift/web/3424" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/shapeshift/web/3424?utm_source=stack-comment).

File: src/features/defi/providers/fox-eth-lp/components/FoxEthLpManager/Withdraw/components/Withdraw.tsx [hunk 1]
```diff
@@ -248,8 +261,8 @@ export const Withdraw: React.FC<WithdrawProps> = ({
             showFiatAmount={true}
             assetIcon={ethAsset.icon}
             assetSymbol={ethAsset.symbol}
-            balance={foxEthLpOpportunity?.underlyingToken0AmountCryptoPrecision ?? undefined}
-            fiatBalance={bnOrZero(foxEthLpOpportunity?.underlyingToken0AmountCryptoPrecision)
+            balance={foxEthLpOpportunity?.underlyingToken0AmountCryptoBaseUnit ?? undefined}
+            fiatBalance={bnOrZero(foxEthLpOpportunity?.underlyingToken0AmountCryptoBaseUnit)
```
[the remark this answers is not in the document]
Comment (0xdef1cafe) [reply src/features/defi/providers/fox-eth-lp/components/FoxEthLpManager/Withdraw/components/Withdraw.tsx:265]: fromBaseUnit?

File: src/state/slices/opportunitiesSlice/selectors.ts [hunk 2]
```diff
@@ -403,25 +387,25 @@ export const selectAggregatedEarnUserStakingOpportunitiesIncludeEmpty =
...
         cryptoAmountBaseUnit: bnOrZero(currentOpportunity.stakedAmountCryptoBaseUnit)
+          .div(bn(10).pow(asset?.precision ?? underlyingAsset?.precision))
           .plus(acc?.stakedAmountCryptoBaseUnit ?? 0)
           .toString(),
-        fiatAmount: bnOrZero(currentOpportunity?.rewardsAmountsCryptoPrecision?.[0])
-          .plus(acc?.rewardsAmountsCryptoPrecision?.[0] ?? 0)
+        fiatAmount: bnOrZero(currentOpportunity?.rewardsAmountsCryptoBaseUnit?.[0])
+          .div(bn(10).pow(asset?.precision ?? underlyingAsset?.precision))
+          .plus(acc?.rewardsAmountsCryptoBaseUnit?.[0] ?? 0)
+          .div(bn(10).pow(asset?.precision ?? underlyingAsset?.precision))
```
Comment (0xdef1cafe) [inline src/state/slices/opportunitiesSlice/selectors.ts:404]: double division? my BODMAS is hurting. add base units and divide once?

Comment (gomesalexandre) [reply src/state/slices/opportunitiesSlice/selectors.ts]: 7d59baac6786376fa14f13bd8117f9b8e457f78

**Review by 0xdef1cafe — APPROVED** [review]
🐌

**Pull request merged**